example/contrib/socks-client/socks/impl/error.ipp

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef SOCKS_IMPL_ERROR_IPP
+#define SOCKS_IMPL_ERROR_IPP
+
+#include <socks/error.hpp>
+
+namespace socks {
+
+namespace detail {
+
+class error_codes : public error_category
+{
+public:
+    const char*
+    name() const noexcept override
+    {
+        return "SOCKS";
+    }
+
+    BOOST_BEAST_DECL
```

Comment:
I would actually like to know if derived overrides need the `__declspec(DLLEXPORT)` (or whatever).